java/server/src/org/openqa/selenium/grid/server/BaseServerOptions.java

```diff
@@ -87,7 +88,11 @@ public URI getExternalUri() {
     int port = getPort();
 
     try {
-      return new URI("http", null, host, port, null, null, null);
```

Comment:
Consider: `return new URI(isSecure() ? "https" : "http", null, host, port, null, null, null)` to emphasise that the only thing that changes is the scheme.

Reply:
Or `getScheme()` as a (private?) method on this class.